# Data Analysis on the Olympics dataset

## The questions that I wish to address from the Olympic dataset are,

### 1. What are the top 10 sports played during the Summer olympics and how have they evolved over a 50-year  period?

### 2. How have the countries fared in terms of medals and which country has had the most success and which countries have had the least success in terms of medals won and how have they been distributed?


# 1. Data Wrangling

## 1.1 Investigating the dataset.


### In order to do so let's explore the data sets first.

In [121]:
import pandas as pd
import numpy as np
from IPython import display
from ipywidgets import interact, widgets
import matplotlib
import matplotlib.pyplot as pp
%matplotlib inline


In [122]:
# Importing the datasets
df_events= pd.read_csv('athlete_events.csv')
df_regions = pd.read_csv('noc_regions.csv')

In [123]:
df_events.head()

ID                      Name Sex   Age  Height  Weight            Team  \
0   1                 A Dijiang   M  24.0   180.0    80.0           China   
1   2                  A Lamusi   M  23.0   170.0    60.0           China   
2   3       Gunnar Nielsen Aaby   M  24.0     NaN     NaN         Denmark   
3   4      Edgar Lindenau Aabye   M  34.0     NaN     NaN  Denmark/Sweden   
4   5  Christine Jacoba Aaftink   F  21.0   185.0    82.0     Netherlands   

   NOC        Games  Year  Season       City          Sport  \
0  CHN  1992 Summer  1992  Summer  Barcelona     Basketball   
1  CHN  2012 Summer  2012  Summer     London           Judo   
2  DEN  1920 Summer  1920  Summer  Antwerpen       Football   
3  DEN  1900 Summer  1900  Summer      Paris     Tug-Of-War   
4  NED  1988 Winter  1988  Winter    Calgary  Speed Skating   

                              Event Medal  
0       Basketball Men's Basketball   NaN  
1      Judo Men's Extra-Lightweight   NaN  
2           Football Men's Football   NaN  
3       Tug-Of-War Men's Tug-Of-War  Gold  
4  Speed Skating Women's 500 metres   NaN

In [124]:
df_events.shape

(271116, 15)

In [125]:
df_events.describe()

ID            Age         Height         Weight  \
count  271116.000000  261642.000000  210945.000000  208241.000000   
mean    68248.954396      25.556898     175.338970      70.702393   
std     39022.286345       6.393561      10.518462      14.348020   
min         1.000000      10.000000     127.000000      25.000000   
25%     34643.000000      21.000000     168.000000      60.000000   
50%     68205.000000      24.000000     175.000000      70.000000   
75%    102097.250000      28.000000     183.000000      79.000000   
max    135571.000000      97.000000     226.000000     214.000000   

                Year  
count  271116.000000  
mean     1978.378480  
std        29.877632  
min      1896.000000  
25%      1960.000000  
50%      1988.000000  
75%      2002.000000  
max      2016.000000

In [126]:
df_events.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 271116 entries, 0 to 271115
Data columns (total 15 columns):
ID        271116 non-null int64
Name      271116 non-null object
Sex       271116 non-null object
Age       261642 non-null float64
Height    210945 non-null float64
Weight    208241 non-null float64
Team      271116 non-null object
NOC       271116 non-null object
Games     271116 non-null object
Year      271116 non-null int64
Season    271116 non-null object
City      271116 non-null object
Sport     271116 non-null object
Event     271116 non-null object
Medal     39783 non-null object
dtypes: float64(3), int64(2), object(10)
memory usage: 31.0+ MB


In [127]:
df_events['Sport'].unique()

array(['Basketball', 'Judo', 'Football', 'Tug-Of-War', 'Speed Skating',
       'Cross Country Skiing', 'Athletics', 'Ice Hockey', 'Swimming',
       'Badminton', 'Sailing', 'Biathlon', 'Gymnastics',
       'Art Competitions', 'Alpine Skiing', 'Handball', 'Weightlifting',
       'Wrestling', 'Luge', 'Water Polo', 'Hockey', 'Rowing', 'Bobsleigh',
       'Fencing', 'Equestrianism', 'Shooting', 'Boxing', 'Taekwondo',
       'Cycling', 'Diving', 'Canoeing', 'Tennis', 'Modern Pentathlon',
       'Figure Skating', 'Golf', 'Softball', 'Archery', 'Volleyball',
       'Synchronized Swimming', 'Table Tennis', 'Nordic Combined',
       'Baseball', 'Rhythmic Gymnastics', 'Freestyle Skiing',
       'Rugby Sevens', 'Trampolining', 'Beach Volleyball', 'Triathlon',
       'Ski Jumping', 'Curling', 'Snowboarding', 'Rugby',
       'Short Track Speed Skating', 'Skeleton', 'Lacrosse', 'Polo',
       'Cricket', 'Racquets', 'Motorboating', 'Military Ski Patrol',
       'Croquet', 'Jeu De Paume', 'Roque', 'Al

In [128]:
df_events.groupby('Sport').nunique()

ID   Name  Sex  Age  Height  Weight  Team  NOC  \
Sport                                                                          
Aeronautics                    1      1    1    1       0       0     1    1   
Alpine Skiing               2735   2734    2   32      46      60   101  101   
Alpinism                      25     25    2   11       0       0     6    6   
Archery                     1113   1113    2   52      44      68   102   98   
Art Competitions            1814   1814    2   68      20      12    51   51   
Athletics                  22071  22053    2   40      65     148   231  227   
Badminton                    811    811    2   25      45      50   101   69   
Baseball                     761    761    1   26      41      56    16   16   
Basketball                  3413   3412    2   25      63      84    66   66   
Basque Pelota                  2      2    1    1       0       0     1    1   
Beach Volleyball             383    383    2   24      47      52    81   41   
Biathlon                    1135   1135    2   27      45      50    54   54   
Bobsleigh                   1694   1694    2   39      47      61   132   54   
Boxing                      5262   5254    2   25      66      90   175  175   
Canoeing                    3206   3204    2   34      53      66   128   99   
Cricket                       24     24    1   11       0       0     2    2   
Croquet                       10     10    2    8       0       0     1    1   
Cross Country Skiing        2400   2400    2   32      50      62    81   81   
Curling                      346    346    2   37      42      51    16   16   
Cycling                     5819   5819    2   33      55      60   150  145   
Diving                      1466   1466    2   33      43      53    82   82   
Equestrianism               2345   2345    2   54      49      52   117   76   
Fencing                     4123   4118    2   46      52      65   101  101   
Figure Skating              1572   1572    2   37      51      55   107   55   
Football                    6161   6161    2   26      51      58   103   92   
Freestyle Skiing             626    626    2   29      46      58    41   41   
Golf                         218    218    2   34      30      43    45   42   
Gymnastics                  4134   4132    2   37      55      64   109   93   
Handball                    2702   2699    2   29      58      70    47   47   
Hockey                      3825   3813    2   30      51      56    47   43   
...                          ...    ...  ...  ...     ...     ...   ...  ...   
Modern Pentathlon            864    864    2   28      41      46    57   57   
Motorboating                  14     14    2    8       1       1     6    2   
Nordic Combined              605    604    1   24      30      37    31   31   
Polo                          87     87    1   27       3       0    17    9   
Racquets                       7      7    1    7       1       0     4    1   
Rhythmic Gymnastics          567    567    1   17      31      31    48   48   
Roque                          4      4    1    3       0       0     1    1   
Rowing                      7687   7684    2   44      68      81   181  101   
Rugby                        155    155    1   18      21      19     8    7   
Rugby Sevens                 299    299    2   19      45      58    14   14   
Sailing                     4480   4480    2   53      53      82   696  120   
Shooting                    4882   4879    2   55      57      93   152  150   
Short Track Speed Skating    444    443    2   23      40      40    35   35   
Skeleton                     146    146    2   30      34      40    28   28   
Ski Jumping                  874    874    2   29      40      49    37   37   
Snowboarding                 567    567    2   26      41      52    35   35   
Softball                     367    367    1   24      36      45    13   13   
Speed Skating               15

#### We see that,
##### 1. There are 271116 rows and 15 columns.
##### 2. There are a total of 66 sports that were played over the past 120 years.
##### 3. There were two seasons, namely Summer and Winter for Summer Olympics and Winter Olympics.


### Now, lets explore the other dataset.

In [129]:
df_regions.head()

NOC       region                 notes
0  AFG  Afghanistan                   NaN
1  AHO      Curacao  Netherlands Antilles
2  ALB      Albania                   NaN
3  ALG      Algeria                   NaN
4  AND      Andorra                   NaN

In [130]:
df_regions.shape

(230, 3)

In [131]:
df_regions.describe()

NOC   region                 notes
count   230      227                    21
unique  230      206                    21
top     SRI  Germany  Refugee Olympic Team
freq      1        4                     1

In [132]:
df_regions.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 230 entries, 0 to 229
Data columns (total 3 columns):
NOC       230 non-null object
region    227 non-null object
notes     21 non-null object
dtypes: object(3)
memory usage: 5.5+ KB


In [133]:
df_regions[(df_regions['region']=='Vietnam') |(df_regions['region'] == 'Yemen')]

NOC   region        notes
220  VIE  Vietnam          NaN
222  VNM  Vietnam          NaN
224  YAR    Yemen  North Yemen
225  YEM    Yemen          NaN
226  YMD    Yemen  South Yemen

### We see that,
##### 1. There are 227 regions that have participated in the Olympics over the years.
##### 2. There are 230 regions and some regions have more than 1 NOC, like Vietnam,Yemen etc. 

### In order to answer the questions above, we need to merge both the datasets and then clean it to get the desired results.

# 2. Data Cleaning

### Before cleaning the data, both the datasets have to be merged.

In [134]:
#Merging the datasets using left join
df=pd.merge(df_events,df_regions, on='NOC', how='left') 

#### Exploring the merged data to find anomalies 

In [135]:
#Checking the merged datasets
df.head()

ID                      Name Sex   Age  Height  Weight            Team  \
0   1                 A Dijiang   M  24.0   180.0    80.0           China   
1   2                  A Lamusi   M  23.0   170.0    60.0           China   
2   3       Gunnar Nielsen Aaby   M  24.0     NaN     NaN         Denmark   
3   4      Edgar Lindenau Aabye   M  34.0     NaN     NaN  Denmark/Sweden   
4   5  Christine Jacoba Aaftink   F  21.0   185.0    82.0     Netherlands   

   NOC        Games  Year  Season       City          Sport  \
0  CHN  1992 Summer  1992  Summer  Barcelona     Basketball   
1  CHN  2012 Summer  2012  Summer     London           Judo   
2  DEN  1920 Summer  1920  Summer  Antwerpen       Football   
3  DEN  1900 Summer  1900  Summer      Paris     Tug-Of-War   
4  NED  1988 Winter  1988  Winter    Calgary  Speed Skating   

                              Event Medal       region notes  
0       Basketball Men's Basketball   NaN        China   NaN  
1      Judo Men's Extra-Lightweight   NaN        China   NaN  
2           Football Men's Football   NaN      Denmark   NaN  
3       Tug-Of-War Men's Tug-Of-War  Gold      Denmark   NaN  
4  Speed Skating Women's 500 metres   NaN  Netherlands   NaN

In [136]:
df.shape

(271116, 17)

In [137]:
df.dtypes

ID          int64
Name       object
Sex        object
Age       float64
Height    float64
Weight    float64
Team       object
NOC        object
Games      object
Year        int64
Season     object
City       object
Sport      object
Event      object
Medal      object
region     object
notes      object
dtype: object

In [138]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 271116 entries, 0 to 271115
Data columns (total 17 columns):
ID        271116 non-null int64
Name      271116 non-null object
Sex       271116 non-null object
Age       261642 non-null float64
Height    210945 non-null float64
Weight    208241 non-null float64
Team      271116 non-null object
NOC       271116 non-null object
Games     271116 non-null object
Year      271116 non-null int64
Season    271116 non-null object
City      271116 non-null object
Sport     271116 non-null object
Event     271116 non-null object
Medal     39783 non-null object
region    270746 non-null object
notes     5039 non-null object
dtypes: float64(3), int64(2), object(12)
memory usage: 37.2+ MB


### Now that both the datasets are mergerd, let's go ahead and clean the dataset.

## 2.1 Removing the unwanted columns and rows


#### There are columns such as notes which are not relavent to out analysis, so let's drop the column.
#### Similarly, for the analysis, regions which have Individual Olympic Athletes,Refugee Olympic Team, Unknown do not play an important role, so these rows can be dropped.

In [139]:
df = df.drop('notes', 1)

In [140]:
# Let's check the dataset now,
df.shape

(271116, 16)

In [141]:
df.dtypes

ID          int64
Name       object
Sex        object
Age       float64
Height    float64
Weight    float64
Team       object
NOC        object
Games      object
Year        int64
Season     object
City       object
Sport      object
Event      object
Medal      object
region     object
dtype: object

#### We see that the notes column has been dropped

#### Now, we'll remove the rows that have Individual Olympic Athletes,Refugee Olympic Team, Unknown values.

In [142]:
df.shape

(271116, 16)

In [143]:
df[(df['NOC'] == 'IOA')].count()

ID        94
Name      94
Sex       94
Age       94
Height    68
Weight    68
Team      94
NOC       94
Games     94
Year      94
Season    94
City      94
Sport     94
Event     94
Medal      5
region    94
dtype: int64

In [144]:
df[(df['NOC'] == 'ROT')].count()

ID        12
Name      12
Sex       12
Age       12
Height    12
Weight    12
Team      12
NOC       12
Games     12
Year      12
Season    12
City      12
Sport     12
Event     12
Medal      0
region     0
dtype: int64

In [145]:
df[(df['NOC'] == 'UNK')].count()

ID        2
Name      2
Sex       2
Age       0
Height    0
Weight    0
Team      2
NOC       2
Games     2
Year      2
Season    2
City      2
Sport     2
Event     2
Medal     0
region    0
dtype: int64

In [146]:
df = df[(df['NOC'] != 'UNK') & (df['NOC'] != 'ROT') & (df['NOC'] != 'IOA')]

In [147]:
df.shape

(271008, 16)

In [148]:
df[(df['NOC'] == 'UNK')].count()

ID        0
Name      0
Sex       0
Age       0
Height    0
Weight    0
Team      0
NOC       0
Games     0
Year      0
Season    0
City      0
Sport     0
Event     0
Medal     0
region    0
dtype: int64

In [149]:
df[(df['NOC'] == 'ROT')].count()

ID        0
Name      0
Sex       0
Age       0
Height    0
Weight    0
Team      0
NOC       0
Games     0
Year      0
Season    0
City      0
Sport     0
Event     0
Medal     0
region    0
dtype: int64

In [150]:
df[(df['NOC'] == 'IOA')].count()

ID        0
Name      0
Sex       0
Age       0
Height    0
Weight    0
Team      0
NOC       0
Games     0
Year      0
Season    0
City      0
Sport     0
Event     0
Medal     0
region    0
dtype: int64

In [151]:
df.shape

(271008, 16)

##### We see that the rows have been removed.


## 2.2 Handling Missing Data

In [152]:
#Checking if any of the column have any null values
df.isna().sum()

ID             0
Name           0
Sex            0
Age         9472
Height     60143
Weight     62847
Team           0
NOC            0
Games          0
Year           0
Season         0
City           0
Sport          0
Event          0
Medal     231230
region       356
dtype: int64

#### We see that Age, Height,Weight,Medal and region have a significant amount of NULL values. Lets take each of these columns one after the other and explore more to clean the dataset.

### (i) Region

In [153]:
#Listing out those rows where region has Null values
df[df["region"].isna()]

ID                             Name Sex   Age  Height  Weight  \
578        332                 Zarinah Abdullah   F  21.0     NaN     NaN   
579        332                 Zarinah Abdullah   F  25.0     NaN     NaN   
2719      1515  Saiyidah Aisyah Mohamed Rafa'ee   F  28.0   173.0    68.0   
7659      4264                   Ang Peng Siong   M  21.0   180.0    75.0   
7660      4264                   Ang Peng Siong   M  21.0   180.0    75.0   
7661      4264                   Ang Peng Siong   M  21.0   180.0    75.0   
7662      4264                   Ang Peng Siong   M  21.0   180.0    75.0   
7663      4264                   Ang Peng Siong   M  25.0   180.0    75.0   
7664      4264                   Ang Peng Siong   M  25.0   180.0    75.0   
7665      4264                   Ang Peng Siong   M  25.0   180.0    75.0   
7666      4264                   Ang Peng Siong   M  25.0   180.0    75.0   
7667      4264                   Ang Peng Siong   M  25.0   180.0    75.0   
21961    11575             Lon bin Mohamed Noor   M   NaN     NaN    55.0   
27339    14199       Mei Yen "Christel" Bouvron   F  15.0   165.0    62.0   
27340    14199       Mei Yen "Christel" Bouvron   F  15.0   165.0    62.0   
27341    14199       Mei Yen "Christel" Bouvron   F  19.0   165.0    62.0   
33525    17291                      Cai Xiao Li   M  28.0   178.0    70.0   
38075    19565                     Chai Hon Yam   M  28.0     NaN     NaN   
38286    19679          Chan Khun Sing "Joseph"   M  39.0   174.0    68.0   
38287    19679          Chan Khun Sing "Joseph"   M  43.0   174.0    68.0   
38291    19682            Kum Seng "Kelly" Chan   M  27.0   180.0    67.0   
38312    19694                Li Yin "Pat" Chan   F  18.0   173.0    64.0   
38313    19694                Li Yin "Pat" Chan   F  18.0   173.0    64.0   
38319    19697               Kum Wah "Roy" Chan   M  16.0     NaN     NaN   
38320    19697               Kum Wah "Roy" Chan   M  16.0     NaN     NaN   
38321    19697               Kum Wah "Roy" Chan   M  16.0     NaN     NaN   
38951    20063             Jung Jun "Mark" Chay   M  18.0   176.0    78.0   
38952    20063             Jung Jun "Mark" Chay   M  18.0   176.0    78.0   
38953    20063             Jung Jun "Mark" Chay   M  22.0   176.0    78.0   
38954    20063             Jung Jun "Mark" Chay   M  22.0   176.0    78.0   
...        ...                              ...  ..   ...     ...     ...   
265500  132855          Wei Ling "Joscelin" Yeo   F  17.0   165.0    56.0   
265501  132855          Wei Ling "Joscelin" Yeo   F  17.0   165.0    56.0   
265502  132855          Wei Ling "Joscelin" Yeo   F  17.0   165.0    56.0   
265503  132855          Wei Ling "Joscelin" Yeo   F  17.0   165.0    56.0   
265504  132855          Wei Ling "Joscelin" Yeo   F  17.0   165.0    56.0   
265505  132855          Wei Ling "Joscelin" Yeo   F  17.0   165.0    56.0   
265506  132855          Wei Ling "Joscelin" Yeo   F  21.0   165.0    56.0   
265507  132855          Wei Ling "Joscelin" Yeo   F  21.0   165.0    56.0   
265508  132855          Wei Ling "Joscelin" Yeo   F  21.0   165.0    56.0   
265509  132855          Wei Ling "Joscelin" Yeo   F  21.0   165.0    56.0   
265510  132855          Wei Ling "Joscelin" Yeo   F  21.0   165.0    56.0   
265511  132855          Wei Ling "Joscelin" Yeo   F  25.0   165.0    56.0   
265512  132855          Wei Ling "Joscelin" Yeo   F  25.0   165.0    56.0   
265513  132856             Wi Jin "Kenneth" Yeo   M  18.0     NaN     NaN   
265514  132856             Wi Jin "Kenneth" Yeo   M  18.0     NaN     NaN   
265515  132856             Wi Jin "Kenneth" Yeo   M  18.0     NaN     NaN   
265516  132856             Wi Jin "Kenneth" Yeo   M  18.0     NaN     NaN   
265517  132857                    Yeo Kian Chye   M  28.0   168.0    60.0   
265518  132857                    Yeo Kian Chye   M  28.0   168.0    60.0   
265777  132986          Yin Yue "Elizabeth" Yin   F  20.0   167.0    70.0   
265778  

In [154]:
#Extracting those rows where the region has Null values and putting it in a new dataset
df1=df[df["region"].isna()] 

In [155]:
#Finding out the unique NOCs where the region has Null values
df1["NOC"].unique()

array(['SGP', 'TUV'], dtype=object)

In [156]:
# Exploring the regions dataframe to check where the missing data has to be filled in

df_regions[(df_regions["NOC"]=='TUV') | (df_regions["NOC"]=='ROT')|(df_regions["NOC"]=='UNK')|(df_regions["NOC"]=='SGP')]

# We see that regions doesn't have SGP 

NOC region                 notes
168  ROT    NaN  Refugee Olympic Team
208  TUV    NaN                Tuvalu
213  UNK    NaN               Unknown

In [157]:
# lets list the rows in the merged dataframe where the NOC is SGP and see what can be inferred from there
df[(df["NOC"]=='SGP')]

ID                             Name Sex   Age  Height  Weight  \
578        332                 Zarinah Abdullah   F  21.0     NaN     NaN   
579        332                 Zarinah Abdullah   F  25.0     NaN     NaN   
2719      1515  Saiyidah Aisyah Mohamed Rafa'ee   F  28.0   173.0    68.0   
7659      4264                   Ang Peng Siong   M  21.0   180.0    75.0   
7660      4264                   Ang Peng Siong   M  21.0   180.0    75.0   
7661      4264                   Ang Peng Siong   M  21.0   180.0    75.0   
7662      4264                   Ang Peng Siong   M  21.0   180.0    75.0   
7663      4264                   Ang Peng Siong   M  25.0   180.0    75.0   
7664      4264                   Ang Peng Siong   M  25.0   180.0    75.0   
7665      4264                   Ang Peng Siong   M  25.0   180.0    75.0   
7666      4264                   Ang Peng Siong   M  25.0   180.0    75.0   
7667      4264                   Ang Peng Siong   M  25.0   180.0    75.0   
21961    11575             Lon bin Mohamed Noor   M   NaN     NaN    55.0   
27339    14199       Mei Yen "Christel" Bouvron   F  15.0   165.0    62.0   
27340    14199       Mei Yen "Christel" Bouvron   F  15.0   165.0    62.0   
27341    14199       Mei Yen "Christel" Bouvron   F  19.0   165.0    62.0   
33525    17291                      Cai Xiao Li   M  28.0   178.0    70.0   
38075    19565                     Chai Hon Yam   M  28.0     NaN     NaN   
38286    19679          Chan Khun Sing "Joseph"   M  39.0   174.0    68.0   
38287    19679          Chan Khun Sing "Joseph"   M  43.0   174.0    68.0   
38291    19682            Kum Seng "Kelly" Chan   M  27.0   180.0    67.0   
38312    19694                Li Yin "Pat" Chan   F  18.0   173.0    64.0   
38313    19694                Li Yin "Pat" Chan   F  18.0   173.0    64.0   
38319    19697               Kum Wah "Roy" Chan   M  16.0     NaN     NaN   
38320    19697               Kum Wah "Roy" Chan   M  16.0     NaN     NaN   
38321    19697               Kum Wah "Roy" Chan   M  16.0     NaN     NaN   
38951    20063             Jung Jun "Mark" Chay   M  18.0   176.0    78.0   
38952    20063             Jung Jun "Mark" Chay   M  18.0   176.0    78.0   
38953    20063             Jung Jun "Mark" Chay   M  22.0   176.0    78.0   
38954    20063             Jung Jun "Mark" Chay   M  22.0   176.0    78.0   
...        ...                              ...  ..   ...     ...     ...   
265500  132855          Wei Ling "Joscelin" Yeo   F  17.0   165.0    56.0   
265501  132855          Wei Ling "Joscelin" Yeo   F  17.0   165.0    56.0   
265502  132855          Wei Ling "Joscelin" Yeo   F  17.0   165.0    56.0   
265503  132855          Wei Ling "Joscelin" Yeo   F  17.0   165.0    56.0   
265504  132855          Wei Ling "Joscelin" Yeo   F  17.0   165.0    56.0   
265505  132855          Wei Ling "Joscelin" Yeo   F  17.0   165.0    56.0   
265506  132855          Wei Ling "Joscelin" Yeo   F  21.0   165.0    56.0   
265507  132855          Wei Ling "Joscelin" Yeo   F  21.0   165.0    56.0   
265508  132855          Wei Ling "Joscelin" Yeo   F  21.0   165.0    56.0   
265509  132855          Wei Ling "Joscelin" Yeo   F  21.0   165.0    56.0   
265510  132855          Wei Ling "Joscelin" Yeo   F  21.0   165.0    56.0   
265511  132855          Wei Ling "Joscelin" Yeo   F  25.0   165.0    56.0   
265512  132855          Wei Ling "Joscelin" Yeo   F  25.0   165.0    56.0   
265513  132856             Wi Jin "Kenneth" Yeo   M  18.0     NaN     NaN   
265514  132856             Wi Jin "Kenneth" Yeo   M  18.0     NaN     NaN   
265515  132856             Wi Jin "Kenneth" Yeo   M  18.0     NaN     NaN   
265516  132856             Wi Jin "Kenneth" Yeo   M  18.0     NaN     NaN   
265517  132857                    Yeo Kian Chye   M  28.0   168.0    60.0   
265518  132857                    Yeo Kian Chye   M  28.0   168.0    60.0   
265777  132986          Yin Yue "Elizabeth" Yin   F  20.0   167.0    70.0   
265778  

#### From the two above codes we see that SGP stands for Singapore, ROT for Refugee Olympic Team, TUV for Tuvalu and UNK for Unkown
#### Now, we will replace each of the NaN values with the respective NOCs
#### In order to do that, first we will replace the NaN values with a random text value.

In [158]:
df['region'].fillna('text', inplace=True)

In [159]:
#Checking the number of rows that have 'text'.

df[df['region']=='text'].shape 

#We see that it matches the number of NaNs as we saw before, i.e, 370

(356, 16)

In [160]:
#Now, we will replace all the 'text' values with the respective NOC values
df.loc[(df['NOC'] == 'SGP'),'region']='Singapore'
df.loc[(df['NOC'] == 'ROT'),'region']='Refugee Olympic Team'
df.loc[(df['NOC'] == 'TUV'),'region']='Tuvalu'
df.loc[(df['NOC'] == 'UNK'),'region']='Unknown'

In [161]:
#Let's now check if there are any rows that have 'text' left.
df[df['region']=='text'].shape

(0, 16)

#### We see that there are none

In [162]:
#Let's now check if there are any rows that have NaN left.
df["region"].isna().sum()

0

In [163]:
df['region'].unique()

array(['China', 'Denmark', 'Netherlands', 'USA', 'Finland', 'Norway',
       'Romania', 'Estonia', 'France', 'Morocco', 'Spain', 'Egypt',
       'Iran', 'Bulgaria', 'Italy', 'Chad', 'Azerbaijan', 'Sudan',
       'Russia', 'Argentina', 'Cuba', 'Belarus', 'Greece', 'Cameroon',
       'Turkey', 'Chile', 'Mexico', 'Nicaragua', 'Hungary', 'Nigeria',
       'Algeria', 'Kuwait', 'Bahrain', 'Pakistan', 'Iraq', 'Syria',
       'Lebanon', 'Qatar', 'Malaysia', 'Germany', 'Canada', 'Ireland',
       'Australia', 'South Africa', 'Eritrea', 'Tanzania', 'Jordan',
       'Tunisia', 'Libya', 'Belgium', 'Djibouti', 'Palestine', 'Comoros',
       'Kazakhstan', 'Brunei', 'India', 'Saudi Arabia', 'Maldives',
       'Ethiopia', 'United Arab Emirates', 'Yemen', 'Indonesia',
       'Philippines', 'Singapore', 'Uzbekistan', 'Kyrgyzstan',
       'Tajikistan', 'Japan', 'Republic of Congo', 'Switzerland',
       'Brazil', 'Monaco', 'Israel', 'Uruguay', 'Sweden',
       'Virgin Islands, US', 'Sri Lanka', 'Armenia'

In [164]:
df['region'].nunique()

206

### (ii) Medal

In [165]:
#Listing out those rows where Medal has Null values
df[df["Medal"].isna()]

ID                                Name Sex   Age  Height  Weight  \
0            1                           A Dijiang   M  24.0   180.0    80.0   
1            2                            A Lamusi   M  23.0   170.0    60.0   
2            3                 Gunnar Nielsen Aaby   M  24.0     NaN     NaN   
4            5            Christine Jacoba Aaftink   F  21.0   185.0    82.0   
5            5            Christine Jacoba Aaftink   F  21.0   185.0    82.0   
6            5            Christine Jacoba Aaftink   F  25.0   185.0    82.0   
7            5            Christine Jacoba Aaftink   F  25.0   185.0    82.0   
8            5            Christine Jacoba Aaftink   F  27.0   185.0    82.0   
9            5            Christine Jacoba Aaftink   F  27.0   185.0    82.0   
10           6                     Per Knut Aaland   M  31.0   188.0    75.0   
11           6                     Per Knut Aaland   M  31.0   188.0    75.0   
12           6                     Per Knut Aaland   M  31.0   188.0    75.0   
13           6                     Per Knut Aaland   M  31.0   188.0    75.0   
14           6                     Per Knut Aaland   M  33.0   188.0    75.0   
15           6                     Per Knut Aaland   M  33.0   188.0    75.0   
16           6                     Per Knut Aaland   M  33.0   188.0    75.0   
17           6                     Per Knut Aaland   M  33.0   188.0    75.0   
18           7                        John Aalberg   M  31.0   183.0    72.0   
19           7                        John Aalberg   M  31.0   183.0    72.0   
20           7                        John Aalberg   M  31.0   183.0    72.0   
21           7                        John Aalberg   M  31.0   183.0    72.0   
22           7                        John Aalberg   M  33.0   183.0    72.0   
23           7                        John Aalberg   M  33.0   183.0    72.0   
24           7                        John Aalberg   M  33.0   183.0    72.0   
25           7                        John Aalberg   M  33.0   183.0    72.0   
26           8  Cornelia "Cor" Aalten (-Strannood)   F  18.0   168.0     NaN   
27           8  Cornelia "Cor" Aalten (-Strannood)   F  18.0   168.0     NaN   
28           9                    Antti Sami Aalto   M  26.0   186.0    96.0   
29          10      Einar Ferdinand "Einari" Aalto   M  26.0     NaN     NaN   
30          11                  Jorma Ilmari Aalto   M  22.0   182.0    76.5   
...        ...                                 ...  ..   ...     ...     ...   
271084  135554                        Bogusaw Zych   M  36.0   182.0    82.0   
271085  135554                        Bogusaw Zych   M  36.0   182.0    82.0   
271086  135555                          Micha Zych   M  23.0   177.0    68.0   
271087  135556          Bogusaw Stanisaw Zychowicz   M  19.0   189.0    80.0   
271088  135556          Bogusaw Stanisaw Zychowicz   M  19.0   189.0    80.0   
271089  135557                        Dominik ycki   M  34.0   192.0    95.0   
271090  135557                        Dominik ycki   M  38.0   192.0    95.0   
271091  135558                  ukasz Tomasz ygado   M  32.0   200.0    89.0   
271092  135559                    Pawe Jan Zygmunt   M  21.0   182.0    79.0   
271093  135559                    Pawe Jan Zygmunt   M  21.0   182.0    79.0   
271094  135559                    Pawe Jan Zygmunt   M  25.0   182.0    79.0   
271095  135559                    Pawe Jan Zygmunt   M  25.0   182.0    79.0   
271096  135559                    Pawe Jan Zygmunt   M  29.0   182.0    79.0   
271097  135559                    Pawe Jan Zygmunt   M  29.0   182.0    79.0   
271098  135559                    Pawe Jan Zygmunt   M  33.0   182.0    79.0   
271099  135560                   Stavroula Zygouri   F  36.0   171.0    63.0   
271100  135561                       Frantiek Zyka   M  26.0     NaN     NaN   
271101  135562                          Milan Zyka   M  24.0   173.0    68.0   
271104  1355

#### We know that not every player can win medals in every event they participate in. Here, we replace every NaN value in the Medal Column with a text - "No Medals Won"

In [166]:
df['Medal'].fillna('No Medal Won', inplace=True)

In [167]:
#Let's now check if there are any rows that have NaN left.
df["Medal"].isna().sum()

0

In [168]:
df.head(5)

ID                      Name Sex   Age  Height  Weight            Team  \
0   1                 A Dijiang   M  24.0   180.0    80.0           China   
1   2                  A Lamusi   M  23.0   170.0    60.0           China   
2   3       Gunnar Nielsen Aaby   M  24.0     NaN     NaN         Denmark   
3   4      Edgar Lindenau Aabye   M  34.0     NaN     NaN  Denmark/Sweden   
4   5  Christine Jacoba Aaftink   F  21.0   185.0    82.0     Netherlands   

   NOC        Games  Year  Season       City          Sport  \
0  CHN  1992 Summer  1992  Summer  Barcelona     Basketball   
1  CHN  2012 Summer  2012  Summer     London           Judo   
2  DEN  1920 Summer  1920  Summer  Antwerpen       Football   
3  DEN  1900 Summer  1900  Summer      Paris     Tug-Of-War   
4  NED  1988 Winter  1988  Winter    Calgary  Speed Skating   

                              Event         Medal       region  
0       Basketball Men's Basketball  No Medal Won        China  
1      Judo Men's Extra-Lightweight  No Medal Won        China  
2           Football Men's Football  No Medal Won      Denmark  
3       Tug-Of-War Men's Tug-Of-War          Gold      Denmark  
4  Speed Skating Women's 500 metres  No Medal Won  Netherlands

#### We see that there are none and the NaN values are  replaces with the text.

### (iii) Age, Height, Weight

In [169]:
#Listing out those rows where Age has Null values
df[df["Age"].isna()]

ID                              Name Sex  Age  Height  Weight  \
147         54             Mohamed Jamshid Abadi   M  NaN     NaN     NaN   
152         58                  Georgi Abadzhiev   M  NaN     NaN     NaN   
153         58                  Georgi Abadzhiev   M  NaN     NaN     NaN   
162         66                   Mohamed Abakkar   M  NaN   156.0    48.0   
212        102                 Sayed Fahmy Abaza   M  NaN     NaN     NaN   
213        102                 Sayed Fahmy Abaza   M  NaN     NaN     NaN   
261        133                         Franz Abb   M  NaN     NaN     NaN   
268        139            George "Ioannis" Abbot   M  NaN     NaN     NaN   
302        163                   Ismail Abdallah   M  NaN     NaN     NaN   
303        163                   Ismail Abdallah   M  NaN     NaN     NaN   
304        163                   Ismail Abdallah   M  NaN     NaN     NaN   
305        163                   Ismail Abdallah   M  NaN     NaN     NaN   
306        163                   Ismail Abdallah   M  NaN     NaN     NaN   
307        163                   Ismail Abdallah   M  NaN     NaN     NaN   
308        163                   Ismail Abdallah   M  NaN     NaN     NaN   
309        163                   Ismail Abdallah   M  NaN     NaN     NaN   
314        167              Ould Lamine Abdallah   M  NaN     NaN     NaN   
320        173               Mohamed Abdel Fatah   M  NaN   170.0    69.0   
323        176        Mahmoud Atter Abdel Fattah   M  NaN     NaN     NaN   
327        179          Ibrahim Saad Abdel Galil   M  NaN   176.0    73.0   
328        179          Ibrahim Saad Abdel Galil   M  NaN   176.0    73.0   
333        183           Abdel Sattar Abdel Hadj   M  NaN     NaN     NaN   
340        186               Mohamed Abdel Hafiz   M  NaN     NaN     NaN   
343        189  Hwad Ibrahim Abdel Hamid Lumomba   M  NaN   162.0    57.0   
344        190               Ibrahim Abdel Hamid   M  NaN     NaN     NaN   
353        194       Moustafa Mounib Abdel Kader   M  NaN   176.0    67.0   
379        209              Ibrahim Abdel Rahman   M  NaN     NaN     NaN   
394        220                Mohamed Abdel Sami   M  NaN     NaN     NaN   
405        224                  Mohamed Abdel-El   M  NaN     NaN     NaN   
409        226         Sanad Bushara Abdel-Nadif   M  NaN     NaN     NaN   
...        ...                               ...  ..  ...     ...     ...   
270062  135107                      Andreas Ziro   M  NaN     NaN     NaN   
270077  135112                  Georgios Zisimos   M  NaN     NaN     NaN   
270078  135112                  Georgios Zisimos   M  NaN     NaN     NaN   
270079  135112                  Georgios Zisimos   M  NaN     NaN     NaN   
270080  135112                  Georgios Zisimos   M  NaN     NaN     NaN   
270263  135198                Dimitrios Zografos   M  NaN     NaN     NaN   
270279  135204                     Erich Zoigner   M  NaN     NaN     NaN   
270280  135204                     Erich Zoigner   M  NaN     NaN     NaN   
270287  135208                     Oscar Zolezzi   M  NaN     NaN     NaN   
270316  135225                         Bla Zoltn   M  NaN     NaN     NaN   
270317  135226  Gyula Zombori (Zanich, Janicsek)   M  NaN     NaN     NaN   
270318  135226  Gyula Zombori (Zanich, Janicsek)   M  NaN     NaN     NaN   
270353  135239                   Khristos Zorbas   M  NaN     NaN     NaN   
270354  135239                   Khristos Zorbas   M  NaN     NaN     NaN   
270355  135239                   Khristos Zorbas   M  NaN     NaN     NaN   
270356  135239                   Khristos Zorbas   M  NaN     NaN     NaN   
270357  135239                   Khristos Zorbas   M  NaN     NaN     NaN   
270358  135239                   Khristos Zorbas   M  NaN     NaN     NaN   
270364  135243                   Aleksandar Zori   M  NaN     NaN     NaN   
270365  135243                   Aleksandar Zori   M  NaN     NaN     NaN   
270429  

In [170]:
#Listing out those rows where Height has Null values
df[df["Height"].isna()]

ID                             Name Sex   Age  Height  Weight  \
2            3              Gunnar Nielsen Aaby   M  24.0     NaN     NaN   
3            4             Edgar Lindenau Aabye   M  34.0     NaN     NaN   
29          10   Einar Ferdinand "Einari" Aalto   M  26.0     NaN     NaN   
35          15             Arvo Ossian Aaltonen   M  22.0     NaN     NaN   
36          15             Arvo Ossian Aaltonen   M  22.0     NaN     NaN   
37          15             Arvo Ossian Aaltonen   M  30.0     NaN     NaN   
38          15             Arvo Ossian Aaltonen   M  30.0     NaN     NaN   
39          15             Arvo Ossian Aaltonen   M  34.0     NaN     NaN   
58          19            Win Valdemar Aaltonen   M  54.0     NaN     NaN   
83          24                 Nils Egil Aaness   M  24.0     NaN     NaN   
84          24                 Nils Egil Aaness   M  24.0     NaN     NaN   
85          24                 Nils Egil Aaness   M  27.0     NaN     NaN   
86          25                 Alf Lied Aanning   M  24.0     NaN     NaN   
88          27                    Johan Aantjes   M  26.0     NaN     NaN   
91          29             Willemien Aardenburg   F  22.0     NaN     NaN   
95          32             Olav Augunson Aarnes   M  23.0     NaN     NaN   
106         38                     Karl Jan Aas   M  20.0     NaN     NaN   
107         39               Lars Thorlaksn Aas   M  33.0     NaN     NaN   
108         39               Lars Thorlaksn Aas   M  33.0     NaN     NaN   
109         39               Lars Thorlaksn Aas   M  33.0     NaN     NaN   
110         40                  Roald Edgar Aas   M  23.0     NaN     NaN   
111         40                  Roald Edgar Aas   M  27.0     NaN     NaN   
112         40                  Roald Edgar Aas   M  27.0     NaN     NaN   
113         40                  Roald Edgar Aas   M  31.0     NaN     NaN   
114         40                  Roald Edgar Aas   M  31.0     NaN     NaN   
115         40                  Roald Edgar Aas   M  31.0     NaN     NaN   
116         41                         Rolf Aas   M  28.0     NaN     NaN   
117         42              Thomas Valentin Aas   M  25.0     NaN     NaN   
124         45                      Hakon Aasns   M  41.0     NaN     NaN   
134         48                 Abdelhak Aatakni   M  24.0     NaN    64.0   
...        ...                              ...  ..   ...     ...     ...   
270915  135481      Jules Alexis "Louis" Zutter   M  30.0     NaN     NaN   
270916  135481      Jules Alexis "Louis" Zutter   M  30.0     NaN     NaN   
270917  135481      Jules Alexis "Louis" Zutter   M  30.0     NaN     NaN   
270919  135483      Andrey Aleksandrovich Zuyev   M  29.0     NaN     NaN   
270933  135487      Vladimir Anatolyevich Zuyev   M  35.0     NaN     NaN   
270944  135490             Irina Lvovna Zuykova   F  34.0     NaN     NaN   
270945  135490             Irina Lvovna Zuykova   F  34.0     NaN     NaN   
270955  135494                     Jacek uawski   M  41.0     NaN     NaN   
270960  135498                     Denis vegelj   M  20.0     NaN     NaN   
270961  135498                     Denis vegelj   M  24.0     NaN     NaN   
270962  135499                Alberts Zvejnieks   M  25.0     NaN     NaN   
270963  135499                Alberts Zvejnieks   M  33.0     NaN     NaN   
270985  135507                      Karel Zvona   M  23.0     NaN     NaN   
270991  135510                 Jan Marcus Zwaan   M  22.0     NaN     NaN   
270992  135511          Johan Gerard "Jo" Zwaan   M  25.0     NaN     NaN   
270993  135511          Johan Gerard "Jo" Zwaan   M  25.0     NaN     NaN   
270998  135513                  Federico Zwanck   M  18.0     NaN     NaN   
270999  135513                  Federico Zwanck   M  18.0     NaN     NaN   
271000  135513                  Federico Zwanck   M  18.0     NaN     NaN   
271003  135515     Adriaan "Adrie" Zwartepoorte   M  19.0     NaN     NaN   
271005  

In [171]:
#Listing out those rows where Weight has Null values
df[df["Weight"].isna()]

ID                                Name Sex   Age  Height  Weight  \
2            3                 Gunnar Nielsen Aaby   M  24.0     NaN     NaN   
3            4                Edgar Lindenau Aabye   M  34.0     NaN     NaN   
26           8  Cornelia "Cor" Aalten (-Strannood)   F  18.0   168.0     NaN   
27           8  Cornelia "Cor" Aalten (-Strannood)   F  18.0   168.0     NaN   
29          10      Einar Ferdinand "Einari" Aalto   M  26.0     NaN     NaN   
35          15                Arvo Ossian Aaltonen   M  22.0     NaN     NaN   
36          15                Arvo Ossian Aaltonen   M  22.0     NaN     NaN   
37          15                Arvo Ossian Aaltonen   M  30.0     NaN     NaN   
38          15                Arvo Ossian Aaltonen   M  30.0     NaN     NaN   
39          15                Arvo Ossian Aaltonen   M  34.0     NaN     NaN   
58          19               Win Valdemar Aaltonen   M  54.0     NaN     NaN   
79          21           Ragnhild Margrethe Aamodt   F  27.0   163.0     NaN   
83          24                    Nils Egil Aaness   M  24.0     NaN     NaN   
84          24                    Nils Egil Aaness   M  24.0     NaN     NaN   
85          24                    Nils Egil Aaness   M  27.0     NaN     NaN   
86          25                    Alf Lied Aanning   M  24.0     NaN     NaN   
88          27                       Johan Aantjes   M  26.0     NaN     NaN   
91          29                Willemien Aardenburg   F  22.0     NaN     NaN   
95          32                Olav Augunson Aarnes   M  23.0     NaN     NaN   
106         38                        Karl Jan Aas   M  20.0     NaN     NaN   
107         39                  Lars Thorlaksn Aas   M  33.0     NaN     NaN   
108         39                  Lars Thorlaksn Aas   M  33.0     NaN     NaN   
109         39                  Lars Thorlaksn Aas   M  33.0     NaN     NaN   
110         40                     Roald Edgar Aas   M  23.0     NaN     NaN   
111         40                     Roald Edgar Aas   M  27.0     NaN     NaN   
112         40                     Roald Edgar Aas   M  27.0     NaN     NaN   
113         40                     Roald Edgar Aas   M  31.0     NaN     NaN   
114         40                     Roald Edgar Aas   M  31.0     NaN     NaN   
115         40                     Roald Edgar Aas   M  31.0     NaN     NaN   
116         41                            Rolf Aas   M  28.0     NaN     NaN   
...        ...                                 ...  ..   ...     ...     ...   
270914  135481         Jules Alexis "Louis" Zutter   M  30.0     NaN     NaN   
270915  135481         Jules Alexis "Louis" Zutter   M  30.0     NaN     NaN   
270916  135481         Jules Alexis "Louis" Zutter   M  30.0     NaN     NaN   
270917  135481         Jules Alexis "Louis" Zutter   M  30.0     NaN     NaN   
270919  135483         Andrey Aleksandrovich Zuyev   M  29.0     NaN     NaN   
270933  135487         Vladimir Anatolyevich Zuyev   M  35.0     NaN     NaN   
270944  135490                Irina Lvovna Zuykova   F  34.0     NaN     NaN   
270945  135490                Irina Lvovna Zuykova   F  34.0     NaN     NaN   
270955  135494                        Jacek uawski   M  41.0     NaN     NaN   
270960  135498                        Denis vegelj   M  20.0     NaN     NaN   
270961  135498                        Denis vegelj   M  24.0     NaN     NaN   
270962  135499                   Alberts Zvejnieks   M  25.0     NaN     NaN   
270963  135499                   Alberts Zvejnieks   M  33.0     NaN     NaN   
270985  135507                         Karel Zvona   M  23.0     NaN     NaN   
270991  135510                    Jan Marcus Zwaan   M  22.0     NaN     NaN   
270992  135511             Johan Gerard "Jo" Zwaan   M  25.0     NaN     NaN   
270993  135511             Johan Gerard "Jo" Zwaan   M  25.0     NaN     NaN   
270998  135513                     Federico Zwanck   M  18.0     NaN     NaN   
270999  1355

#### The age,height and weight columns have missing data, but these data cannot be averaged or cannot be filled in based on any other columns as each of the Players have different height, weight and age.These columns cannot be dropped, instead I'll be replacing the missing data with a string, "Data not recorded"

In [172]:
#Now, let's replace all the NaN values in Age,Height and Weight fields.
df['Age'].fillna('Data not recorded.', inplace=True)
df['Height'].fillna('Data not recorded.', inplace=True)
df['Weight'].fillna('Data not recorded.', inplace=True)

In [173]:
# Let's now check again if any of the fields have NaN values left.
df.isna().sum()

ID        0
Name      0
Sex       0
Age       0
Height    0
Weight    0
Team      0
NOC       0
Games     0
Year      0
Season    0
City      0
Sport     0
Event     0
Medal     0
region    0
dtype: int64

In [174]:
df.head()

ID                      Name Sex Age              Height  \
0   1                 A Dijiang   M  24                 180   
1   2                  A Lamusi   M  23                 170   
2   3       Gunnar Nielsen Aaby   M  24  Data not recorded.   
3   4      Edgar Lindenau Aabye   M  34  Data not recorded.   
4   5  Christine Jacoba Aaftink   F  21                 185   

               Weight            Team  NOC        Games  Year  Season  \
0                  80           China  CHN  1992 Summer  1992  Summer   
1                  60           China  CHN  2012 Summer  2012  Summer   
2  Data not recorded.         Denmark  DEN  1920 Summer  1920  Summer   
3  Data not recorded.  Denmark/Sweden  DEN  1900 Summer  1900  Summer   
4                  82     Netherlands  NED  1988 Winter  1988  Winter   

        City          Sport                             Event         Medal  \
0  Barcelona     Basketball       Basketball Men's Basketball  No Medal Won   
1     London           Judo      Judo Men's Extra-Lightweight  No Medal Won   
2  Antwerpen       Football           Football Men's Football  No Medal Won   
3      Paris     Tug-Of-War       Tug-Of-War Men's Tug-Of-War          Gold   
4    Calgary  Speed Skating  Speed Skating Women's 500 metres  No Medal Won   

        region  
0        China  
1        China  
2      Denmark  
3      Denmark  
4  Netherlands

In [175]:
#Extract the cleaned dataset 
df.to_csv(r'D:\DCU\1st Semester\CA 682\Assignment 1 Data Vis\Datasets\Olympics.csv')

#### Now that the missing data, NULL values and unwanted columns and rows are handled, we can go ahead and analyse and visualise the data.

# 3. Data Analysis and Visualization

## 1. What are the top 10 sports played during the Summer olympics and how have they evolved over a 50 year  period?

### Now, lets visualize the data to answer the first question. 

#### In order to do so, let's first divide the dataset so that we can take out the rows that have only Summer Olympics

In [176]:
df_summer = df[df['Season'] == 'Summer']

In [177]:
df_summer.head(4)

ID                  Name Sex Age              Height              Weight  \
0   1             A Dijiang   M  24                 180                  80   
1   2              A Lamusi   M  23                 170                  60   
2   3   Gunnar Nielsen Aaby   M  24  Data not recorded.  Data not recorded.   
3   4  Edgar Lindenau Aabye   M  34  Data not recorded.  Data not recorded.   

             Team  NOC        Games  Year  Season       City       Sport  \
0           China  CHN  1992 Summer  1992  Summer  Barcelona  Basketball   
1           China  CHN  2012 Summer  2012  Summer     London        Judo   
2         Denmark  DEN  1920 Summer  1920  Summer  Antwerpen    Football   
3  Denmark/Sweden  DEN  1900 Summer  1900  Summer      Paris  Tug-Of-War   

                          Event         Medal   region  
0   Basketball Men's Basketball  No Medal Won    China  
1  Judo Men's Extra-Lightweight  No Medal Won    China  
2       Football Men's Football  No Medal Won  Denmark  
3   Tug-Of-War Men's Tug-Of-War          Gold  Denmark

In [178]:
df_summer.shape

(222445, 16)

In [179]:
df_summer.groupby('Year').nunique()

ID   Name  Sex  Age  Height  Weight  Team  NOC  Games  Year  Season  \
Year                                                                           
1896    176    176    1   21      14      13    18   12      1     1       1   
1900   1224   1220    2   49      25      28   190   31      1     1       1   
1904    650    650    2   50      28      42    79   15      1     1       1   
1906    841    841    2   40      27      40    52   21      1     1       1   
1908   2024   2024    2   47      38      58    73   22      1     1       1   
1912   2407   2407    2   52      40      63   101   28      1     1       1   
1920   2676   2675    2   50      45      70    72   29      1     1       1   
1924   3256   3256    2   60      45      72    90   45      1     1       1   
1928   3247   3246    2   63      49      70    67   46      1     1       1   
1932   1922   1922    2   67      49      66    59   47      1     1       1   
1936   4484   4482    2   62      50      81   105   49      1     1       1   
1948   4402   4402    2   67      47      83   111   59      1     1       1   
1952   4932   4931    2   52      55     109   133   69      1     1       1   
1956   3347   3346    2   51      60     100   122   72      1     1       1   
1960   5352   5348    2   54      65     123   186   84      1     1       1   
1964   5137   5134    2   49      67     116   168   93      1     1       1   
1968   5558   5552    2   54      69     117   112  112      1     1       1   
1972   7114   7105    2   52      73     123   139  121      1     1       1   
1976   6073   6070    2   49      77     128    92   92      1     1       1   
1980   5259   5252    2   44      81     129    80   80      1     1       1   
1984   6798   6791    2   47      78     123   140  140      1     1       1   
1988   8454   8443    2   48      83     131   177  159      1     1       1   
1992   9328   9322    2   51      82     140   211  168      1     1       1   
1996  10339  10324    2   48      81     153   246  197      1     1       1   
2000  10643  10635    2   45      83     160   242  199      1     1       1   
2004  10557  10537    2   44      82     155   260  201      1     1       1   
2008  10899  10880    2   51      83     142   292  204      1     1       1   
2012  10513  10498    2   48      78     133   244  204      1     1       1   
2016  11160  11155    2   48      79     124   247  205      1     1       1   

      City  Sport  Event  Medal  region  
Year                                     
1896     1      9     43      4      12  
1900     1     20     90      4      31  
1904     1     18     95      4      14  
1906     1     13     74      4      20  
1908     1     24    109      4      22  
1912     1     17    107      4      28  
1920     1     25    158      4      29  
1924     1     20    131      4      45  
1928     1     17    122      4      46  
1932     1     18    131      4      47  
1936     1     24    150      4      49  
1948     1     20    153      4      59  
1952     1     19    149      4      67  
1956     2     19    151      4      71  
1960     1     19    150      4      83  
1964     1     21    163      4      93  
1968     1     20    172      4     111  
1972     1     23    193      4     120  
1976     1     23    198      4      91  
1980     1     23    203      4      80  
1984     1     25    221      4     139  
1988     1     27    237      4     156  
1992     1     29    257      4     167  
1996     1     31    271      4     196  
2000     1     34    300      4     198  
2004     1     34    301      4     200  
2008     1     34    302      4     203  
2012     1     32    302      4     203  
2016     1     34    306      4     204

In [180]:
#The top 10 popular sports played in all the countries over the years.
df_summer.groupby(['Sport'])['ID'].count().sort_values(ascending=False).head(10)

Sport
Athletics     38605
Gymnastics    26707
Swimming      23181
Shooting      11429
Cycling       10851
Fencing       10733
Rowing        10593
Wrestling      7145
Football       6745
Sailing        6585
Name: ID, dtype: int64

In [181]:
#Extracting the rows which have the top 10 sports 
df1=df_summer[(df_summer['Sport']=='Athletics') | (df_summer['Sport']=='Gymnastics') | (df_summer['Sport']=='Swimming') | (df_summer['Sport']=='Shooting') |(df_summer['Sport']=='Cycling') | (df_summer['Sport']=='Fencing') | (df_summer['Sport']=='Rowing') | (df_summer['Sport']=='Wrestling') | (df_summer['Sport']=='Football') | (df_summer['Sport']=='Sailing')]

In [182]:
df1.head(3)

ID                                Name Sex Age              Height  \
2    3                 Gunnar Nielsen Aaby   M  24  Data not recorded.   
26   8  Cornelia "Cor" Aalten (-Strannood)   F  18                 168   
27   8  Cornelia "Cor" Aalten (-Strannood)   F  18                 168   

                Weight         Team  NOC        Games  Year  Season  \
2   Data not recorded.      Denmark  DEN  1920 Summer  1920  Summer   
26  Data not recorded.  Netherlands  NED  1932 Summer  1932  Summer   
27  Data not recorded.  Netherlands  NED  1932 Summer  1932  Summer   

           City      Sport                                   Event  \
2     Antwerpen   Football                 Football Men's Football   
26  Los Angeles  Athletics            Athletics Women's 100 metres   
27  Los Angeles  Athletics  Athletics Women's 4 x 100 metres Relay   

           Medal       region  
2   No Medal Won      Denmark  
26  No Medal Won  Netherlands  
27  No Medal Won  Netherlands

In [183]:
#Now, lets group the above data by year and subsequently by sport and count the values and assign to a new dataframe
df2=df1.groupby(['Year','Sport'])['ID'].count()

In [184]:
df2.head(3)

Year  Sport    
1896  Athletics    106
      Cycling       41
      Fencing       15
Name: ID, dtype: int64

In [185]:
#Extract the new dataset 
df2.to_csv(r'D:\DCU\1st Semester\CA 682\Assignment 1 Data Vis\Datasets\Summer_Olympics.csv',header= False)

In [186]:
#Read the dataset to a new dataframe
df_new = pd.read_csv('Summer_Olympics.csv')

In [187]:
df_new.head()

1896   Athletics  106
0  1896     Cycling   41
1  1896     Fencing   15
2  1896  Gymnastics   97
3  1896    Shooting   65
4  1896    Swimming   18

In [188]:
#Assigning new column names to the dataset
df_new.columns = ['Year','Sport','Count_Per_Year'] 

In [189]:
df_new.head(3)

Year       Sport  Count_Per_Year
0  1896     Cycling              41
1  1896     Fencing              15
2  1896  Gymnastics              97

## Plotting the Graph


#### In order to plot the graph let's define a function to plot interactive graphs for the Summer Olympics.

In [190]:
import seaborn as sns

def newplot_summer(Year):
    data = df_new[df_new.Year == Year]
    pp.figure(figsize=(9.5,9.5))
    sns.set_style("whitegrid")
    sns.scatterplot(x='Year',y='Count_Per_Year',hue=df_new['Sport'],data=data,s=150,edgecolor="k")
    
    
    pp.axis(ymin=0,ymax=3000,xmin=1960,xmax=2020)
    pp.xticks(np.arange(1960, 2020, step=4))
    pp.yticks(np.arange(0, 3000, step=100))
    pp.xlabel('Year',fontsize=14, labelpad=10)
    pp.ylabel('Number of Participants',fontsize=14, labelpad=15)
    pp.title('How sports at Summer Olympics have evolved over the past 50 years.',fontsize=16,y=1.02)
    pp.tick_params(labelsize=13, pad=6)
    

#### Now, to make an interactive visualization, we will add an interact() fucntion to see how the games have evolved over 50 years.

In [191]:
interact(newplot_summer,Year=widgets.IntSlider(min=1964,max=2016,step=4))

interactive(children=(IntSlider(value=1964, description='Year', max=2016, min=1964, step=4), Output()), _dom_c…

<function __main__.newplot_summer(Year)>

# ANALYSIS


#### 1. We see from the plot that the top 10 sports which were played in the last 50 years in the Summer Olympics were, Cycling,Fencing,Gymnastics,Shooting,Swimming,Wrestling,Athletics,Football,Rowing and Sailing.

#### 2. The top three games that were played over the past 50 years were Gymnastics,Athletics and Swimming. 

#### 3. The sport which has seen a drastic change in the number of people playing the sport is Athletics. In 1964, there were a little over 1300 athletes and the number rose to 2500 in the year 2016. The number of swimmers in the same period have almost doubled from about 700 swimmers in the year 1964 to about 1600 swimmers in 2016. Gymnastics was the most popular game played in the year 1964 with about 1500 Gymnasts participating in the Olympics but the numbers dwindled down to a little under 900 in the same period,.Gymnastics is the only game played in the top 3 which saw a decrease in the number of participants.

#### 4. The rest of the top 10 games played did not see a big variation in the number of participants over the years. The numbers for the remaining games ranged between 200 to 700 participants over the 50-year period.

 ## 2. How have the countires fared in terms of medals and which country has had the most success and which countries have had the least success in terms of medals won and how have they been distributed?


### In order to answer this question, we will first take those rows where players have won medals.

In [192]:
# Extracting only those rows where the players have won medals.
df_medal = df[~df['Medal'].str.contains('No Medal Won')]

In [193]:
df_medal.head(4)

ID                      Name Sex Age              Height  \
3    4      Edgar Lindenau Aabye   M  34  Data not recorded.   
37  15      Arvo Ossian Aaltonen   M  30  Data not recorded.   
38  15      Arvo Ossian Aaltonen   M  30  Data not recorded.   
40  16  Juhamatti Tapio Aaltonen   M  28                 184   

                Weight            Team  NOC        Games  Year  Season  \
3   Data not recorded.  Denmark/Sweden  DEN  1900 Summer  1900  Summer   
37  Data not recorded.         Finland  FIN  1920 Summer  1920  Summer   
38  Data not recorded.         Finland  FIN  1920 Summer  1920  Summer   
40                  85         Finland  FIN  2014 Winter  2014  Winter   

         City       Sport                                   Event   Medal  \
3       Paris  Tug-Of-War             Tug-Of-War Men's Tug-Of-War    Gold   
37  Antwerpen    Swimming  Swimming Men's 200 metres Breaststroke  Bronze   
38  Antwerpen    Swimming  Swimming Men's 400 metres Breaststroke  Bronze   
40      Sochi  Ice Hockey             Ice Hockey Men's Ice Hockey  Bronze   

     region  
3   Denmark  
37  Finland  
38  Finland  
40  Finland

### To plot a choropleth map, we need country codes. In order to do so, we need to add a new column to the dataset.

In [194]:
df_medal['region'].unique()

array(['Denmark', 'Finland', 'Norway', 'Netherlands', 'France', 'Italy',
       'Spain', 'Azerbaijan', 'Russia', 'Belarus', 'Cameroon', 'USA',
       'Hungary', 'Greece', 'Australia', 'Iran', 'Canada', 'Pakistan',
       'Uzbekistan', 'Tajikistan', 'Japan', 'Ethiopia', 'Switzerland',
       'Germany', 'Sweden', 'South Africa', 'Turkey', 'Bulgaria', 'Egypt',
       'UK', 'Jordan', 'Romania', 'Puerto Rico', 'Mexico', 'Ghana',
       'Morocco', 'New Zealand', 'Argentina', 'Cuba', 'Uruguay', 'Poland',
       'Czech Republic', 'Nigeria', 'Brazil', 'Austria', 'Lithuania',
       'Indonesia', 'Chile', 'Ukraine', 'Uganda', 'Syria', 'Qatar',
       'Kuwait', 'Saudi Arabia', 'United Arab Emirates', 'Croatia',
       'Armenia', 'Serbia', 'Niger', 'India', 'Belgium', 'Algeria',
       'Jamaica', 'Trinidad', 'Colombia', 'Luxembourg', 'Botswana',
       'Tunisia', 'South Korea', 'North Korea', 'China', 'Guyana',
       'Israel', 'Kazakhstan', 'Georgia', 'Kenya', 'Iceland', 'Haiti',
       'Malaysia'

In [195]:
# Adding a new column for country codes.
df_medal['Country Code'] = 'text'

C:\Users\bidis\Anaconda3\lib\site-packages\ipykernel_launcher.py:2: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy



In [196]:
df_medal.head()

ID                      Name Sex Age              Height  \
3    4      Edgar Lindenau Aabye   M  34  Data not recorded.   
37  15      Arvo Ossian Aaltonen   M  30  Data not recorded.   
38  15      Arvo Ossian Aaltonen   M  30  Data not recorded.   
40  16  Juhamatti Tapio Aaltonen   M  28                 184   
41  17   Paavo Johannes Aaltonen   M  28                 175   

                Weight            Team  NOC        Games  Year  Season  \
3   Data not recorded.  Denmark/Sweden  DEN  1900 Summer  1900  Summer   
37  Data not recorded.         Finland  FIN  1920 Summer  1920  Summer   
38  Data not recorded.         Finland  FIN  1920 Summer  1920  Summer   
40                  85         Finland  FIN  2014 Winter  2014  Winter   
41                  64         Finland  FIN  1948 Summer  1948  Summer   

         City       Sport                                   Event   Medal  \
3       Paris  Tug-Of-War             Tug-Of-War Men's Tug-Of-War    Gold   
37  Antwerpen    Swimming  Swimming Men's 200 metres Breaststroke  Bronze   
38  Antwerpen    Swimming  Swimming Men's 400 metres Breaststroke  Bronze   
40      Sochi  Ice Hockey             Ice Hockey Men's Ice Hockey  Bronze   
41     London  Gymnastics  Gymnastics Men's Individual All-Around  Bronze   

     region Country Code  
3   Denmark         text  
37  Finland         text  
38  Finland         text  
40  Finland         text  
41  Finland         text

In [197]:
# Now, we will fill in the Country Code column with the respective Alpha 3 Country Codes.
df_medal.loc[(df['region'] == 'Denmark'),'Country Code']='DNK'
df_medal.loc[(df['region'] == 'Finland'),'Country Code']='FIN'
df_medal.loc[(df['region'] == 'Norway'),'Country Code']='NOR'
df_medal.loc[(df['region'] == 'Netherlands'),'Country Code']='NLD'
df_medal.loc[(df['region'] == 'France'),'Country Code']='FRA'
df_medal.loc[(df['region'] == 'Italy'),'Country Code']='ITA'
df_medal.loc[(df['region'] == 'Spain'),'Country Code']='ESP'
df_medal.loc[(df['region'] == 'Azerbaijan'),'Country Code']='AZE'
df_medal.loc[(df['region'] == 'Russia'),'Country Code']='RUS'
df_medal.loc[(df['region'] == 'Belarus'),'Country Code']='BLR'
df_medal.loc[(df['region'] == 'Cameroon'),'Country Code']='CMR'
df_medal.loc[(df['region'] == 'USA'),'Country Code']='USA'
df_medal.loc[(df['region'] == 'Hungary'),'Country Code']='HUN'
df_medal.loc[(df['region'] == 'Greece'),'Country Code']='GRC'
df_medal.loc[(df['region'] == 'Australia'),'Country Code']='AUS'
df_medal.loc[(df['region'] == 'Iran'),'Country Code']='IRN'
df_medal.loc[(df['region'] == 'Canada'),'Country Code']='CAN'
df_medal.loc[(df['region'] == 'Pakistan'),'Country Code']='PAK'
df_medal.loc[(df['region'] == 'Uzbekistan'),'Country Code']='UZB'
df_medal.loc[(df['region'] == 'Tajikistan'),'Country Code']='TJK'
df_medal.loc[(df['region'] == 'Japan'),'Country Code']='JPN'
df_medal.loc[(df['region'] == 'Ethiopia'),'Country Code']='ETH'
df_medal.loc[(df['region'] == 'Switzerland'),'Country Code']='CHE'
df_medal.loc[(df['region'] == 'Germany'),'Country Code']='DEU'
df_medal.loc[(df['region'] == 'Sweden'),'Country Code']='SWE'
df_medal.loc[(df['region'] == 'South Africa'),'Country Code']='ZAF'
df_medal.loc[(df['region'] == 'Turkey'),'Country Code']='TUR'
df_medal.loc[(df['region'] == 'Bulgaria'),'Country Code']='BGR'
df_medal.loc[(df['region'] == 'Egypt'),'Country Code']='EGY'
df_medal.loc[(df['region'] == 'UK'),'Country Code']='GBR'
df_medal.loc[(df['region'] == 'Jordan'),'Country Code']='JOR'
df_medal.loc[(df['region'] == 'Romania'),'Country Code']='ROU'
df_medal.loc[(df['region'] == 'Puerto Rico'),'Country Code']='PRI'
df_medal.loc[(df['region'] == 'Mexico'),'Country Code']='MEX'
df_medal.loc[(df['region'] == 'Ghana'),'Country Code']='GHA'
df_medal.loc[(df['region'] == 'Morocco'),'Country Code']='MAR'
df_medal.loc[(df['region'] == 'New Zealand'),'Country Code']='NZL'
df_medal.loc[(df['region'] == 'Argentina'),'Country Code']='ARG'
df_medal.loc[(df['region'] == 'Cuba'),'Country Code']='CUB'
df_medal.loc[(df['region'] == 'Uruguay'),'Country Code']='URY'
df_medal.loc[(df['region'] == 'Poland'),'Country Code']='POL'
df_medal.loc[(df['region'] == 'Czech Republic'),'Country Code']='CZE'
df_medal.loc[(df['region'] == 'Nigeria'),'Country Code']='NGA'
df_medal.loc[(df['region'] == 'Brazil'),'Country Code']='BRA'
df_medal.loc[(df['region'] == 'Austria'),'Country Code']='AUT'
df_medal.loc[(df['region'] == 'Lithuania'),'Country Code']='LTU'
df_medal.loc[(df['region'] == 'Indonesia'),'Country Code']='IDN'
df_medal.loc[(df['region'] == 'Chile'),'Country Code']='CHL'
df_medal.loc[(df['region'] == 'Ukraine'),'Country Code']='UKR'
df_medal.loc[(df['region'] == 'Uganda'),'Country Code']='UGA'
df_medal.loc[(df['region'] == 'Syria'),'Country Code']='SYR'
df_medal.loc[(df['region'] == 'Qatar'),'Country Code']='QAT'
df_medal.loc[(df['region'] == 'Kuwait'),'Country Code']='KWT'
df_medal.loc[(df['region'] == 'Saudi Arabia'),'Country Code']='SAU'
df_medal.loc[(df['region'] == 'United Arab Emirates'),'Country Code']='ARE'
df_medal.loc[(df['region'] == 'Croatia'),'Country Code']='HRV'
df_medal.loc[(df['region'] == 'Armenia'),'Country Code']='ARM'
df_medal.loc[(df['region'] == 'Serbia'),'Country Code']='SRB'
df_medal.loc[(df['region'] == 'Niger'),'Country Code']='NER'
df_medal.loc[(df['region'] == 'India'),'Country Code']='IND'
df_medal.loc[(df['region'] == 'Belgium'),'Country Code']='BEL'
df_medal.loc[(df['region'] == 'Algeria'),'Country Code']='DZA'
df_medal.loc[(df['region'] == 'Jamaica'),'Country Code']='JAM'
df_medal.loc[(df['region'] == 'Trinidad'),'Country Code']='TTO'
df_medal.loc[(df['region'] == 'Colombia'),'Country Code']='COL'
df_medal.loc[(df['region'] == 'Luxembourg'),'Country Code']='LUX'
df_medal.loc[(df['region'] == 'Botswana'),'Country Code']='BWA'
df_medal.loc[(df['region'] == 'Tunisia'),'Country Code']='TUN'
df_medal.loc[(df['region'] == 'South Korea'),'Country Code']='KOR'
df_medal.loc[(df['region'] == 'North Korea'),'Country Code']='PRK'
df_medal.loc[(df['region'] == 'China'),'Country Code']='CHN'
df_medal.loc[(df['region'] == 'Guyana'),'Country Code']='GUY'
df_medal.loc[(df['region'] == 'Israel'),'Country Code']='ISR'
df_medal.loc[(df['region'] == 'Kazakhstan'),'Country Code']='KAZ'
df_medal.loc[(df['region'] == 'Georgia'),'Country Code']='GEO'
df_medal.loc[(df['region'] == 'Kenya'),'Country Code']='KEN'
df_medal.loc[(df['region'] == 'Iceland'),'Country Code']='ISL'
df_medal.loc[(df['region'] == 'Haiti'),'Country Code']='HTI'
df_medal.loc[(df['region'] == 'Malaysia'),'Country Code']='MYS'
df_medal.loc[(df['region'] == 'Iraq'),'Country Code']='IRQ'
df_medal.loc[(df['region'] == 'Slovakia'),'Country Code']='SVK'
df_medal.loc[(df['region'] == 'Bahamas'),'Country Code']='BHS'
df_medal.loc[(df['region'] == 'Paraguay'),'Country Code']='PRY'
df_medal.loc[(df['region'] == 'Montenegro'),'Country Code']='MNE'
df_medal.loc[(df['region'] == 'Ireland'),'Country Code']='IRL'
df_medal.loc[(df['region'] == 'Portugal'),'Country Code']='PRT'
df_medal.loc[(df['region'] == 'Guatemala'),'Country Code']='GTM'
df_medal.loc[(df['region'] == 'Tanzania'),'Country Code']='TZA'
df_medal.loc[(df['region'] == 'Lebanon'),'Country Code']='LBN'
df_medal.loc[(df['region'] == 'Kyrgyzstan'),'Country Code']='KGZ'
df_medal.loc[(df['region'] == 'Latvia'),'Country Code']='LVA'
df_medal.loc[(df['region'] == 'Curacao'),'Country Code']='CUW'
df_medal.loc[(df['region'] == 'Venezuela'),'Country Code']='VEN'
df_medal.loc[(df['region'] == 'Thailand'),'Country Code']='THA'
df_medal.loc[(df['region'] == 'Togo'),'Country Code']='TGO'
df_medal.loc[(df['region'] == 'Peru'),'Country Code']='PER'
df_medal.loc[(df['region'] == 'Estonia'),'Country Code']='EST'
df_medal.loc[(df['region'] == 'Slovenia'),'Country Code']='SVN'
df_medal.loc[(df['region'] == 'Nepal'),'Country Code']='NPL'
df_medal.loc[(df['region'] == 'Taiwan'),'Country Code']='TWN'
df_medal.loc[(df['region'] == 'Zimbabwe'),'Country Code']='ZWE'
df_medal.loc[(df['region'] == 'Mongolia'),'Country Code']='MNG'
df_medal.loc[(df['region'] == 'Moldova'),'Country Code']='MDA'
df_medal.loc[(df['region'] == 'Ivory Coast'),'Country Code']='CIV'
df_medal.loc[(df['region'] == 'Fiji'),'Country Code']='FJI'
df_medal.loc[(df['region'] == 'Senegal'),'Country Code']='SEN'
df_medal.loc[(df['region'] == 'Dominican Republic'),'Country Code']='DOM'
df_medal.loc[(df['region'] == 'Philippines'),'Country Code']='PHL'
df_medal.loc[(df['region'] == 'Singapore'),'Country Code']='SGP'
df_medal.loc[(df['region'] == 'Namibia'),'Country Code']='NAM'
df_medal.loc[(df['region'] == 'Liechtenstein'),'Country Code']='LIE'
df_medal.loc[(df['region'] == 'Bermuda'),'Country Code']='BMU'
df_medal.loc[(df['region'] == 'Vietnam'),'Country Code']='VNM'
df_medal.loc[(df['region'] == 'Virgin Islands, US'),'Country Code']='VIR'
df_medal.loc[(df['region'] == 'Macedonia'),'Country Code']='MKD'
df_medal.loc[(df['region'] == 'Sudan'),'Country Code']='SDN'
df_medal.loc[(df['region'] == 'Bahrain'),'Country Code']='BHR'
df_medal.loc[(df['region'] == 'Grenada'),'Country Code']='GRD'
df_medal.loc[(df['region'] == 'Sri Lanka'),'Country Code']='LKA'
df_medal.loc[(df['region'] == 'Mauritius'),'Country Code']='MUS'
df_medal.loc[(df['region'] == 'Kosovo'),'Country Code']='XKX'
df_medal.loc[(df['region'] == 'Cyprus'),'Country Code']='CYP'
df_medal.loc[(df['region'] == 'Panama'),'Country Code']='PAN'
df_medal.loc[(df['region'] == 'Zambia'),'Country Code']='ZMB'
df_medal.loc[(df['region'] == 'Monaco'),'Country Code']='MCO'
df_medal.loc[(df['region'] == 'Mozambique'),'Country Code']='MOZ'
df_medal.loc[(df['region'] == 'Suriname'),'Country Code']='SUR'
df_medal.loc[(df['region'] == 'Afghanistan'),'Country Code']='AFG'
df_medal.loc[(df['region'] == 'Burundi'),'Country Code']='BDI'
df_medal.loc[(df['region'] == 'Gabon'),'Country Code']='GAB'
df_medal.loc[(df['region'] == 'Ecuador'),'Country Code']='ECU'
df_medal.loc[(df['region'] == 'Costa Rica'),'Country Code']='CRI'
df_medal.loc[(df['region'] == 'Djibouti'),'Country Code']='DJI'
df_medal.loc[(df['region'] == 'Eritrea'),'Country Code']='ERI'
df_medal.loc[(df['region'] == 'Barbados'),'Country Code']='BRB'
df_medal.loc[(df['region'] == 'Tonga'),'Country Code']='TON'




C:\Users\bidis\Anaconda3\lib\site-packages\pandas\core\indexing.py:543: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy



In [198]:
df_medal['Country Code'].nunique()

136

In [199]:
df_medal.shape

(39778, 17)

In [200]:
# Grouping the dataset as per medals won by each country in the olympics.
df_medal.groupby(['region','Country Code'])['Medal'].count()

region                Country Code
Afghanistan           AFG                2
Algeria               DZA               17
Argentina             ARG              274
Armenia               ARM               16
Australia             AUS             1349
Austria               AUT              450
Azerbaijan            AZE               44
Bahamas               BHS               40
Bahrain               BHR                3
Barbados              BRB                1
Belarus               BLR              139
Belgium               BEL              468
Bermuda               BMU                1
Botswana              BWA                1
Brazil                BRA              475
Bulgaria              BGR              342
Burundi               BDI                2
Cameroon              CMR               22
Canada                CAN             1352
Chile                 CHL               32
China                 CHN              993
Colombia              COL               28
Costa Rica         

In [201]:
# Assigning the output to a new dataframe
df_medal_new=df_medal.groupby(['region','Country Code'])['Medal'].count()

In [202]:
# Printing the new dataset to a csv.
df_medal_new.to_csv(r'D:\DCU\1st Semester\CA 682\Assignment 1 Data Vis\Datasets\Medals.csv',header= False)

In [203]:
#Reading the csv file
dataframe = pd.read_csv('Medals.csv',header=None)
dataframe.head()

0    1     2
0  Afghanistan  AFG     2
1      Algeria  DZA    17
2    Argentina  ARG   274
3      Armenia  ARM    16
4    Australia  AUS  1349

In [204]:
# Assigning new Column Names.
dataframe.columns = ['Country','Country Code','Medal Count'] 

In [205]:
# Printing out the dataset.
dataframe.head()

Country Country Code  Medal Count
0  Afghanistan          AFG            2
1      Algeria          DZA           17
2    Argentina          ARG          274
3      Armenia          ARM           16
4    Australia          AUS         1349

#### Now, that we have the number of medals per country, let's further dive in and find out how many medals of each type did the countries that participated in Olympics win, i.e., the number of Gold,Silver and Bronze medal that each country has won.

In [206]:
# Let's look at the main dataset
df_medal.shape

(39778, 17)

In [207]:
df_medal.head(3)

ID                  Name Sex Age              Height              Weight  \
3    4  Edgar Lindenau Aabye   M  34  Data not recorded.  Data not recorded.   
37  15  Arvo Ossian Aaltonen   M  30  Data not recorded.  Data not recorded.   
38  15  Arvo Ossian Aaltonen   M  30  Data not recorded.  Data not recorded.   

              Team  NOC        Games  Year  Season       City       Sport  \
3   Denmark/Sweden  DEN  1900 Summer  1900  Summer      Paris  Tug-Of-War   
37         Finland  FIN  1920 Summer  1920  Summer  Antwerpen    Swimming   
38         Finland  FIN  1920 Summer  1920  Summer  Antwerpen    Swimming   

                                     Event   Medal   region Country Code  
3              Tug-Of-War Men's Tug-Of-War    Gold  Denmark          DNK  
37  Swimming Men's 200 metres Breaststroke  Bronze  Finland          FIN  
38  Swimming Men's 400 metres Breaststroke  Bronze  Finland          FIN

In [208]:
# Now, we will group the regions by the medals that they won.
df_medal.groupby(['region','Medal'])['Medal'].count()

region                Medal 
Afghanistan           Bronze       2
Algeria               Bronze       8
                      Gold         5
                      Silver       4
Argentina             Bronze      91
                      Gold        91
                      Silver      92
Armenia               Bronze       9
                      Gold         2
                      Silver       5
Australia             Bronze     522
                      Gold       368
                      Silver     459
Austria               Bronze     156
                      Gold       108
                      Silver     186
Azerbaijan            Bronze      25
                      Gold         7
                      Silver      12
Bahamas               Bronze      15
                      Gold        14
                      Silver      11
Bahrain               Bronze       1
                      Gold         1
                      Silver       1
Barbados              Bronze       1
Belarus  

In [209]:
df_indv_medal=df_medal.groupby(['region','Medal'])['Medal'].count()

In [210]:
# Printing the data into a new csv file.
df_indv_medal.to_csv(r'D:\DCU\1st Semester\CA 682\Assignment 1 Data Vis\Datasets\Indv_Medal.csv',header= False)

In [211]:
#Reading the new csv file
df1= pd.read_csv('Indv_Medal.csv',header=None)
df1.head(3)

0       1  2
0  Afghanistan  Bronze  2
1      Algeria  Bronze  8
2      Algeria    Gold  5

In [212]:
df1.columns = ['Country','Medal Type','Individual Medal Count'] 

In [213]:
df1.head(3)

Country Medal Type  Individual Medal Count
0  Afghanistan     Bronze                       2
1      Algeria     Bronze                       8
2      Algeria       Gold                       5

In [214]:
#Now, we will merge dataset and df1 to make a dataset which will have all the information we need to plot the graph for the second question
ddf = pd.merge(dataframe,df1, on='Country', how='outer') 

In [215]:
ddf.head()

Country Country Code  Medal Count Medal Type  Individual Medal Count
0  Afghanistan          AFG            2     Bronze                       2
1      Algeria          DZA           17     Bronze                       8
2      Algeria          DZA           17       Gold                       5
3      Algeria          DZA           17     Silver                       4
4    Argentina          ARG          274     Bronze                      91

In [216]:
#ddf['Individual Medal Count'] = ddf['Individual Medal Count'].apply(str)


In [217]:
ddf.dtypes

Country                   object
Country Code              object
Medal Count                int64
Medal Type                object
Individual Medal Count     int64
dtype: object

In [218]:
ddf['Medal Count']=ddf['Medal Count'].astype(str)
ddf['Individual Medal Count']=ddf['Individual Medal Count'].astype(str)
ddf.dtypes

Country                   object
Country Code              object
Medal Count               object
Medal Type                object
Individual Medal Count    object
dtype: object

In [219]:
ddf['text'] = ddf['Medal Type'] + ' ' + ddf['Individual Medal Count']
ddf.head()

Country Country Code Medal Count Medal Type Individual Medal Count  \
0  Afghanistan          AFG           2     Bronze                      2   
1      Algeria          DZA          17     Bronze                      8   
2      Algeria          DZA          17       Gold                      5   
3      Algeria          DZA          17     Silver                      4   
4    Argentina          ARG         274     Bronze                     91   

        text  
0   Bronze 2  
1   Bronze 8  
2     Gold 5  
3   Silver 4  
4  Bronze 91

In [220]:
print (ddf.groupby(['Country'])['text'].apply(' '.join).reset_index())

                  Country                               text
0             Afghanistan                           Bronze 2
1                 Algeria           Bronze 8 Gold 5 Silver 4
2               Argentina        Bronze 91 Gold 91 Silver 92
3                 Armenia           Bronze 9 Gold 2 Silver 5
4               Australia     Bronze 522 Gold 368 Silver 459
5                 Austria     Bronze 156 Gold 108 Silver 186
6              Azerbaijan         Bronze 25 Gold 7 Silver 12
7                 Bahamas        Bronze 15 Gold 14 Silver 11
8                 Bahrain           Bronze 1 Gold 1 Silver 1
9                Barbados                           Bronze 1
10                Belarus        Bronze 71 Gold 24 Silver 44
11                Belgium      Bronze 173 Gold 98 Silver 197
12                Bermuda                           Bronze 1
13               Botswana                           Silver 1
14                 Brazil     Bronze 191 Gold 109 Silver 175
15               Bulgari

In [221]:
ddf.head()

Country Country Code Medal Count Medal Type Individual Medal Count  \
0  Afghanistan          AFG           2     Bronze                      2   
1      Algeria          DZA          17     Bronze                      8   
2      Algeria          DZA          17       Gold                      5   
3      Algeria          DZA          17     Silver                      4   
4    Argentina          ARG         274     Bronze                     91   

        text  
0   Bronze 2  
1   Bronze 8  
2     Gold 5  
3   Silver 4  
4  Bronze 91

In [222]:
ddf['text'] = ddf['Medal Type'] + ' ' + ddf['Individual Medal Count']

In [223]:
ddf.head()

Country Country Code Medal Count Medal Type Individual Medal Count  \
0  Afghanistan          AFG           2     Bronze                      2   
1      Algeria          DZA          17     Bronze                      8   
2      Algeria          DZA          17       Gold                      5   
3      Algeria          DZA          17     Silver                      4   
4    Argentina          ARG         274     Bronze                     91   

        text  
0   Bronze 2  
1   Bronze 8  
2     Gold 5  
3   Silver 4  
4  Bronze 91

In [224]:
ddf1=ddf.groupby(['Country'])['text'].apply(' '.join).reset_index()

In [225]:
ddf1.head()

Country                            text
0  Afghanistan                        Bronze 2
1      Algeria        Bronze 8 Gold 5 Silver 4
2    Argentina     Bronze 91 Gold 91 Silver 92
3      Armenia        Bronze 9 Gold 2 Silver 5
4    Australia  Bronze 522 Gold 368 Silver 459

In [226]:
ddf1.to_csv(r'D:\DCU\1st Semester\CA 682\Assignment 1 Data Vis\Datasets\Text_to_display.csv')

In [227]:
ddf2 = pd.read_csv('Text_to_display.csv')

In [228]:
ddf2.head()

Unnamed: 0      Country                            text
0           0  Afghanistan                        Bronze 2
1           1      Algeria        Bronze 8 Gold 5 Silver 4
2           2    Argentina     Bronze 91 Gold 91 Silver 92
3           3      Armenia        Bronze 9 Gold 2 Silver 5
4           4    Australia  Bronze 522 Gold 368 Silver 459

In [229]:
#ddf2.columns = ['ID','Country','text'] 

In [230]:
ddf2=ddf2.drop("Unnamed: 0",axis=1)

In [231]:
ddf2.head()

Country                            text
0  Afghanistan                        Bronze 2
1      Algeria        Bronze 8 Gold 5 Silver 4
2    Argentina     Bronze 91 Gold 91 Silver 92
3      Armenia        Bronze 9 Gold 2 Silver 5
4    Australia  Bronze 522 Gold 368 Silver 459

In [232]:
dataframe1=pd.merge(dataframe,ddf2, on='Country', how='left') 

In [233]:
dataframe1.head()

Country Country Code  Medal Count                            text
0  Afghanistan          AFG            2                        Bronze 2
1      Algeria          DZA           17        Bronze 8 Gold 5 Silver 4
2    Argentina          ARG          274     Bronze 91 Gold 91 Silver 92
3      Armenia          ARM           16        Bronze 9 Gold 2 Silver 5
4    Australia          AUS         1349  Bronze 522 Gold 368 Silver 459

In [234]:
# Now, we will group the countries by the number of players and write that to a csv.
df.groupby(['region'])['Name'].count().to_csv(r'D:\DCU\1st Semester\CA 682\Assignment 1 Data Vis\Datasets\Total_Players.csv')

C:\Users\bidis\Anaconda3\lib\site-packages\ipykernel_launcher.py:2: FutureWarning:

The signature of `Series.to_csv` was aligned to that of `DataFrame.to_csv`, and argument 'header' will change its default value from False to True: please pass an explicit value to suppress this warning.



In [235]:
#Now that we have the number of players who have taken part, we will now merge this with the medals dataset.
#To do that, let's read the dataset first.
players = pd.read_csv('Total_players.csv',header= None)
players.head()

0    1
0     Afghanistan  126
1         Albania   70
2         Algeria  551
3  American Samoa   37
4         Andorra  169

In [236]:
#Now, let's assign appropriate header to the columns.
players.columns = ['Country','Player Count'] 
players.head()

Country  Player Count
0     Afghanistan           126
1         Albania            70
2         Algeria           551
3  American Samoa            37
4         Andorra           169

In [237]:
#Now, we will merge medal and players to make a dataset which will have all the information we need to plot the graph for the third question
dataframe1 = pd.merge(dataframe1,players, on='Country', how='outer') 
dataframe1.head()

Country Country Code  Medal Count                            text  \
0  Afghanistan          AFG          2.0                        Bronze 2   
1      Algeria          DZA         17.0        Bronze 8 Gold 5 Silver 4   
2    Argentina          ARG        274.0     Bronze 91 Gold 91 Silver 92   
3      Armenia          ARM         16.0        Bronze 9 Gold 2 Silver 5   
4    Australia          AUS       1349.0  Bronze 522 Gold 368 Silver 459   

   Player Count  
0           126  
1           551  
2          3297  
3           221  
4          7724

#### Now, that we have all the columns we need, let's plot a choropleth map depicting the total number of medals and the number of each type of medal that every country that has participated in Olympics has won so far.

In [238]:
from plotly.offline import download_plotlyjs, init_notebook_mode, plot, iplot

In [239]:
data = dict(type = 'choropleth',
           locations = dataframe1['Country Code'],
           z = dataframe1['Medal Count'],
           text = 'Country : '+ dataframe1['Country']  + '<br>Individual Medals Won : '+dataframe1['text'] + '<br>No. of Participants : '+dataframe1['Player Count'].astype(str),
           colorbar = {'title': 'Medal Count'},
           autocolorscale=False,
           reversescale=False,
           colorscale = 'sunsetdark' 
           )

layout = dict(title = 'Medals won by countries and total number of participants in Olympics over 120 years.',
              geo = dict(showframe = False,
                       projection = {'type' : 'natural earth'}))

In [240]:
fig = dict(data = data ,layout = layout)

plot(fig, filename = 'medals.html')

'medals.html'

## ANALYSIS

#### 1. We see from the choropleth that the United States of America has won the most medals as compared to any other countries in the world,with 2638 Gold medals,1641 Silver Medals, and 1358 Bronze medals, brings up their tally to 5637 medals over a period of 120 years in the Olympics.

#### 2. Russia and Germany have more than 3500 medals with over 1200 Gold medals each. Among the Asian countries China,Japan and South Korea have been the most successful in the Olympics and Sri Lanka,Nepal and Tajikistan are among the countries that have won the least medals among the countries which have won medals in the last 120 years.

#### 3. We can also infer from the plot that, Europe has been the most successful continent in terms of the medals won and Africa has been the least.

#### 4. Among Oceania, Australia and New Zealand have won the most medals, 1349 and 228 respectively.